### PR TITLE
Add ASRC simulator framework to sln_voice

### PR DESCRIPTION
**Pull request by shuchitak**
https://xmosjira.atlassian.net/browse/AP-127

Simulator framework described https://xmosjira.atlassian.net/wiki/spaces/~701212200/pages/3878944769/ASRC+USB+in+I2S+out+simulation and https://xmosjira.atlassian.net/wiki/spaces/~701212200/pages/3875897371/ASRC+I2S+in+USB+out+simulation

Not covered in this PR:
I want to reuse the avg buffer level and rate calculation functions in the actual ASRC demo application for the simulation app as well. However, this will require a bit of change in the actual ASRC demo app source files which I don't want to touch as part of this PR. I'll do this as a separate change once this is merged.

### CODE REVIEW

File: test/asrc_sim/python/plot_csv.py [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+    if showplot == True:
+        plt.show()
+
+
+# Usage: python ../plot_csv.py <log> <plotname> <1 or 2>
+
+def get_args():
+    parser = argparse.ArgumentParser("Script to plot the stdout when running the usb_in_i2s_out or i2s_in_sb_out application")
+    parser.add_argument("input_file", type=str, help="stdout when running the usb_in_i2s_out or i2s_in_sb_out application dumped into a log file")
+    parser.add_argument("num_cols", type=int, help="No. of comma separated entries per line in the stdout of the usb_in_i2s_out or i2s_in_sb_out application run")
```
Comment (danielpieczko) [inline test/asrc_sim/python/plot_csv.py:99]:
```suggestion
    parser = argparse.ArgumentParser("Script to plot the stdout when running the usb_in_i2s_out or i2s_in_usb_out application")
    parser.add_argument("input_file", type=str, help="stdout when running the usb_in_i2s_out or i2s_in_usb_out application dumped into a log file")
    parser.add_argument("num_cols", type=int, help="No. of comma separated entries per line in the stdout of the usb_in_i2s_out or i2s_in_usb_out application run")
```

File: test/asrc_sim/run.sh [hunk 2]
```diff
@@ -0,0 +1,58 @@
+#!/bin/bash
```
Comment (xhuw) [inline test/asrc_sim/run.sh:1]: Could do with a comment explaining the purpose of this file

**Review by xhuw — APPROVED** [review]
Haven't looked in too much detail at the code. But I can see the structure is clear and mostly well commented. I think you have done enough here that a future person could figure it out if they needed to.
It's a cool simulator. And you used it to solve the problem, so approved 😄

**Review by danielpieczko — APPROVED** [review]

File: Jenkinsfile [hunk 3]
```diff
@@ -25,22 +25,38 @@ pipeline {
...
             steps {
                 checkout scm
                 sh 'git submodule update --init --recursive --depth 1 --jobs \$(nproc)'
             }
         }
+        stage('ASRC Simulator') {
+            steps {
+                withTools(params.TOOLS_VERSION) {
+                    dir("test/asrc_sim") {
+                        sh "pyenv install -s $PYTHON_VERSION"
```
[the remark this answers is not in the document]
Comment (shuchitak) [reply Jenkinsfile:50]: xmos_jenkins_shared_library is still at v0.20.0 in the Jenkinsfile so createVenv is not supported.

File: test/asrc_sim/src/app_i2s_in_usb_out/pi_control.c [hunk 4]
```diff
@@ -0,0 +1,70 @@
+// Copyright 2023 XMOS LIMITED.
+// This Software is subject to the terms of the XMOS Public Licence: Version 1.
+#include <stdint.h>
+#include <stdbool.h>
+#include <stdio.h>
+#include <string.h>
+#include "pi_control.h"
+
+
+typedef int32_t sw_pll_q24_t; // Type for 15.16 signed fixed point
```
[the remark this answers is not in the document]
Comment (shuchitak) [reply test/asrc_sim/src/app_i2s_in_usb_out/pi_control.c:10]: It used to be 15.16. I'll update the comment.

**Pull request merged**